laravel/app/Http/Middleware/CheckPublished.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Contracts\Auth\Guard;
+use App\Models\Slot;
+
+class CheckPublished
+{
+    /**
+     * The Guard implementation.
+     *
+     * @var Guard
+     */
+    protected $auth;
+
+    /**
+     * Create a new filter instance.
+     *
+     * @param  Guard  $auth
+     * @return void
+     */
+    public function __construct(Guard $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $model_name)
+    {
+        $event = $request->{$model_name};
+        if($request->{$model_name} === 'department') {
+            $event = $request->{$model_name}->event;
+        }
+        if($request->{$model_name} === 'shift') {
+            $event = $request->{$model_name}->event;
+        }
+        if($request->{$model_name} === 'schedule') {
+            $event = $request->{$model_name}->shift->event;
+        }
+        if($request->{$model_name} === 'slot') {
+            $event = $request->{$model_name}->schedule->shift->event;
+        }
+
+        $is_published = ($event->published_at !== null);
+        $is_admin= $this->auth->user()->hasRole('admin');
```

Comment:
Only because the middleware checks lead or admin if it's not published. I couldn't find any documentation or any edge functionality that would allow me to disregard "isAdmin" or "isLead" if the event was published. And loosely defined closure chains can be a pain if they have dependent behavior, so I opted to make it a tidy class.